### PR TITLE
Update docs for 3.8

### PR DESCRIPTION
**Pull request by Jjagg**
This adds docs for
- creating a project
- list of target platforms
- migrating to 3.8 for existing games (I didn't add this to the table of contents)

I also updated tool naming in the docs:
- 2MGFX -> MGFXC
- Pipeline Tool -> MGCB Editor

I did not make changes to the existing getting started guide in this PR. I'm working on redoing it completely, partly based on MrGrak's [Getting Started Guide](https://github.com/MrGrak/Monogame-Getting-Started).

### CODE REVIEW

File: Documentation/platforms.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+- PlayStation Vita
+- Xbox One
+- Nintendo Switch
+- Google Stadia
+
+There are different implementations of MonoGame that we call target platforms (or just platforms).
+The platforms mostly correspond to the systems MonoGame supports but some platforms
+support multiple systems. Platforms for systems that require registration such as PlayStation
+and Xbox are not publicly available. To get access to those platforms, contact Tom Spilman
+at ???. MonoGame documentation for closed platforms is available in their respective repositories.
```
Comment (Jjagg) [inline Documentation/platforms.md:20]: @tomspilman What should I put here?

File: Documentation/platforms.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+## DesktopGL
+
+**Supported Systems**: Windows, Mac, Linux
+**NuGet Package**: MonoGame.Framework.DesktopGL
+**Template ID**: mgdesktopgl
+
+DesktopGL uses SDL for windowing. OpenGL is used for graphics and
+OpenAL Soft for audio. DesktopGL supports Windows (Vista or up),
+Mac (version ???) and Linux.
```
Comment (Jjagg) [inline Documentation/platforms.md:48]: @mrhelmut I don't know what the minimum macOS version we support is

File: Documentation/platforms.md (hunk 1 above)
Comment (mrhelmut) [reply Documentation/platforms.md:20]: The safest is to indicate users to contact their console account manager (which will verify their registration and loop back to Tom to confirm).

**Comment by mrhelmut** [comment]
Cool!

I've submitted a PR to your branch with suggestions and things I've been working on.

File: Documentation/platforms.md (hunk 2 above)
Comment (mrhelmut) [reply Documentation/platforms.md:48]: From the API perspective, none specifically.
.Net Core however, has a High Sierra (macOS 10.13) requirement as of .Net Core 3.

**Comment by mrhelmut** [comment]
Forget to mention that MrGrak's guide looks awesome.

File: Documentation/create_project.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+MonoGame distributes templates to set up a new project in two ways:
+
+- For Visual Studio
+- For .NET Core CLI
```
Comment (Jjagg) [inline Documentation/create_project.md:4]: We should add VS for Mac instructions too. 
@harry-cpp?

File: Documentation/requirements.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+## For desktop platforms
+
+MonoGame requires a .NET Core SDK (3.1 or up) installation.
+
+You can either install it [independently](https://dotnet.microsoft.com/download/dotnet-core), or by selecting the .NET Core payload when installing Visual Studio 2019 (version 15.4 and up required).
+
+If you are targeting WindowsDX, you are also going to need [the DirectX June 2010 runtime](https://www.microsoft.com/en-us/download/details.aspx?id=8109) for audio and gamepads to work properly.
+
+When it comes to IDE, Visual Studio 2019, Visual Studio Code, and Visual Studio 2019 for Mac are supported (alternatively, you can work directly from the CLI with your code editor of choice).
```
Comment (Jjagg) [inline Documentation/requirements.md:12]: We should maybe reword this.

File: Documentation/create_project.md (hunk 3 above)
Comment (harry-cpp) [reply Documentation/create_project.md:4]: Yup.

File: Documentation/requirements.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
+
+Depending on the [platform](Platforms.md) that you are targeting, MonoGame has different sets of requirements.
+
+## For desktop platforms
+
+MonoGame requires a .NET Core SDK (3.1 or up) installation.
+
+You can either install it [independently](https://dotnet.microsoft.com/download/dotnet-core), or by selecting the .NET Core payload when installing Visual Studio 2019 (version 15.4 and up required).
```
Comment (jnoyola) [inline Documentation/requirements.md:8]: [nit] I believe the terminology VS uses is "workload", not "payload".

I've written a lot of docs for people to set up VS 😛

**Comment by Jjagg** [comment]
The docs built with this PR are hosted at https://docs.monogame.net. Check it out :) 
Should be good to go!

**Comment by SimonDarksideJ** [comment]
Thanks for the update @Jjagg , great stuff 🗡️

**Comment by Jjagg** [comment]
@SimonDarksideJ We set up the doc build automation and docs.monogame.net now has the docs from the develop branch. The setting up pages have been replaced (though no building from source instructions yet).

**Comment by tomspilman** [comment]
@Jjagg merged!
